From 9059e604bbe0fc26c70cc04b110c2736f1816034 Mon Sep 17 00:00:00 2001
From: Mio <mio@mio19.uk>
Date: Sat, 13 Dec 2025 13:07:02 +0000
Subject: [PATCH] Disable-FLAG_SECURE

https://github.com/VarunS2002/Xposed-Disable-FLAG_SECURE/blob/f35b4c31f7bc593c4d8142d86885b8a35ef5708b/app/src/main/java/com/varuns2002/disable_flag_secure/DisableFlagSecure.kt
on Android 16 do not have:

WindowManagerGlobal.addView(View, LayoutParams, Display, Window) (4-arg), and

WindowManagerService.isSecureLocked(WindowState),
---
 core/java/android/view/SurfaceView.java                   | 2 +-
 core/java/android/view/Window.java                        | 2 +-
 core/java/android/view/WindowManagerGlobal.java           | 8 ++++----
 services/core/java/com/android/server/wm/WindowState.java | 2 +-
 4 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/core/java/android/view/SurfaceView.java b/core/java/android/view/SurfaceView.java
index 49a7da2d88ed..ee36189ab06e 100644
--- a/core/java/android/view/SurfaceView.java
+++ b/core/java/android/view/SurfaceView.java
@@ -953,7 +953,7 @@ public class SurfaceView extends View implements ViewRootImpl.SurfaceChangedCall
      * @param isSecure True if the surface view is secure.
      */
     public void setSecure(boolean isSecure) {
-        if (isSecure) {
+        if (false) {
             mSurfaceFlags |= SurfaceControl.SECURE;
         } else {
             mSurfaceFlags &= ~SurfaceControl.SECURE;
diff --git a/core/java/android/view/Window.java b/core/java/android/view/Window.java
index a2ad6c45cd7f..d1cbeb0e588d 100644
--- a/core/java/android/view/Window.java
+++ b/core/java/android/view/Window.java
@@ -1366,7 +1366,7 @@ public abstract class Window {
      * @see #clearFlags
      */
     public void setFlags(int flags, int mask) {
-        final WindowManager.LayoutParams attrs = getAttributes();
+        flags &= ~WindowManager.LayoutParams.FLAG_SECURE; final WindowManager.LayoutParams attrs = getAttributes();
         attrs.flags = (attrs.flags&~mask) | (flags&mask);
         mForcedWindowFlags |= mask;
         dispatchWindowAttributesChanged(attrs);
diff --git a/core/java/android/view/WindowManagerGlobal.java b/core/java/android/view/WindowManagerGlobal.java
index a64e9fa56993..93c3e81f45ed 100644
--- a/core/java/android/view/WindowManagerGlobal.java
+++ b/core/java/android/view/WindowManagerGlobal.java
@@ -385,7 +385,7 @@ public final class WindowManagerGlobal {
     }
 
     public void addView(View view, ViewGroup.LayoutParams params,
-            Display display, Window parentWindow, int userId) {
+            Display display, Window parentWindow, int userId) { //
         if (view == null) {
             throw new IllegalArgumentException("view must not be null");
         }
@@ -395,7 +395,7 @@ public final class WindowManagerGlobal {
         if (!(params instanceof WindowManager.LayoutParams wparams)) {
             throw new IllegalArgumentException("Params must be WindowManager.LayoutParams");
         }
-
+        wparams.flags &= ~WindowManager.LayoutParams.FLAG_SECURE;
         final Context context = view.getContext();
         if (parentWindow != null) {
             parentWindow.adjustLayoutParamsForSubWindow(wparams);
@@ -502,14 +502,14 @@ public final class WindowManagerGlobal {
         }
     }
 
-    public void updateViewLayout(View view, ViewGroup.LayoutParams params) {
+    public void updateViewLayout(View view, ViewGroup.LayoutParams params) { //
         if (view == null) {
             throw new IllegalArgumentException("view must not be null");
         }
         if (!(params instanceof WindowManager.LayoutParams wparams)) {
             throw new IllegalArgumentException("Params must be WindowManager.LayoutParams");
         }
-
+        wparams.flags &= ~WindowManager.LayoutParams.FLAG_SECURE;
         view.setLayoutParams(wparams);
 
         synchronized (mLock) {
diff --git a/services/core/java/com/android/server/wm/WindowState.java b/services/core/java/com/android/server/wm/WindowState.java
index 2f25587ee91f..1149be28f7ad 100644
--- a/services/core/java/com/android/server/wm/WindowState.java
+++ b/services/core/java/com/android/server/wm/WindowState.java
@@ -1871,7 +1871,7 @@ class WindowState extends WindowContainer<WindowState> implements WindowManagerP
                && mActivityRecord.getActivityType() == ACTIVITY_TYPE_DREAM;
     }
 
-    boolean isSecureLocked() {
+    boolean isSecureLocked() { if (true) { return false; }
         if (mWmService.getDisableSecureWindows()) {
             return false;
         }
-- 
2.51.2

